app/Http/Controllers/Api/InventoryCollectionController.php

```diff
@@ -0,0 +1,58 @@
+<?php namespace App\Http\Controllers\Api;
+
+use App\Http\Requests;
+use App\Http\Controllers\Controller;
+use App\Item;
+
+use Illuminate\Support\Facades\Response;
+use Illuminate\Support\Facades\Input;
+use Illuminate\Http\Request;
+
+class InventoryCollectionController extends Controller {
+
+	/**
+	 * Display a listing of the resource.
+	 *
+	 * @param  string $query
+	 * @return all queried $items
+	 */
+	public function index($query)
+	{
+		//$items = Item::all();
+
+		if(!is_null($query)) {
+			
+			$items = Item::where('asset_tag', $query)->orWhere('name', $query)
+			->orWhere('funding_source', $query)->orWhere('model', $query)
+			->orWhere('cpu', $query)->orWhere('ram', $query)->orWhere('os', $query)
+			->orWhere('administrator_flag', $query)->orWhere('teacher_flag', $query)
+			->orWhere('student_flag', $query)->orWhere('institution_flag', $query)->get();
+
+			return $items;
+
+		} else {
+			return Response::json("Bad Request", 400);
```

Comment:
I think if `$query` is null (i.e., they didn't include the query string parameter) then return everything.